libs/sdk-core/src/breez_services.rs

```diff
@@ -917,13 +917,18 @@ impl BreezServices {
         Ok(())
     }
 
-    /// Connects to the selected LSP peer. If none is selected, this selects the first one from [`list_lsps`] and persists the selection.
+    /// Connects to the selected LSP peer.
+    /// This validates if the selected LSP is still in [`list_lsps`].
+    /// If not or no LSP is selected, it selects the first LSP in [`list_lsps`].
     async fn connect_lsp_peer(&self, node_pubkey: String) -> SdkResult<()> {
-        // Sets the LSP id, if not already set
-        if self.persister.get_lsp_id()?.is_none() {
-            if let Some(lsp) = self.lsp_api.list_lsps(node_pubkey).await?.first().cloned() {
-                self.persister.set_lsp_id(lsp.id)?;
-            }
+        let lsps = self.lsp_api.list_lsps(node_pubkey).await?;
```

Comment:
This works but has a disadvantage of making this network request on every call of this method.
Below we are calling `lsp_info` which already cals `list_lsps` and return None in case the lsp is not in the list.
Perhaps an optimistic approach where only re-selecting if `lsp_info` return `None` is more efficient?

Reply:
Agreed, it not terribly efficient. The lsp_info call is also used to validate the node state exists. 
I think I can rewrite it so it makes 1 network request, instead of min 1 / max 2 network requests.